src/views/Checkout.vue

```diff
@@ -147,6 +160,73 @@ export default class Checkout extends Vue {
         this.hasBalances = true;
     }
 
+    @Watch('height')
+    private logHeightChange(height: number, oldHeight: number) {
+        console.debug(`Got height: ${height} (was ${oldHeight})`);
+    }
+
+    private async tapFaucet() {
+        const amount = await Faucet.getDispenseAmount();
+        console.log(`tap faucet for ${this.firstAddress.userFriendlyAddress}`);
+        const network = (this.$refs.network as Network);
+        try {
+            await Faucet.tap(this.firstAddress.userFriendlyAddress, null); // currently no captcha required in testnet
+
+            this.isTappingFaucet = true;
+
+            // wait for updated balance
+            const checkForArrivedBalance = async () => {
+                this.getBalances(true);
+                const balances: Map<string, number> = await network.getBalances(
+                    [this.firstAddress.userFriendlyAddress],
+                );
+                const newBalance = [ ...balances.values() ][0];
+                console.log('check balance');
+                if (this.firstAddress.balance && newBalance > this.firstAddress.balance) {
+                    this.firstAddress.balance = newBalance; // still referenced in wallet's accounts map
+                    const updatedWallet = this.wallets[0];
+                    // Update IndexedDB
+                    await WalletStore.Instance.put(updatedWallet);
+                    // Update Vuex
+                    this.$addWallet(updatedWallet);
+                    this.isTappingFaucet = false;
+                }
```

Comment:
Why the duplicate balance fetch via `this.getBalances(true)` and `network.getBalances`?
Also `this.getBalances` invokes `network.connectPico` which disconnects the old network client.
Might be that this could interfere with the `network.getBalances` call?

I'd use only `this.getBalance` (and also await it) which removes the double fetch and necessity to update the store here.
The check whether the balance changes should then happen against a stored old value.